datasources/src/main/scala/com/normation/plugins/datasources/Scheduler.scala

```diff
@@ -108,7 +110,7 @@ class DataSourceScheduler(
 
     for {
       s <- schedule
-      p <- prog.repeat(s).provide(clock).unit
+      p <- prog.schedule(s).provide(clock).unit
```

Comment:
We need to use `schedule` to make `never` work, because `repeat` always execute at least once while `schedule` starts by trying to understand the schedule. That's also why we need to use `Schedule.once.andThen` now, we don't get the one free from `repeat`. Not that the semantic is better now, `repeat` always repeats, and there is no magical things done for `Schedule.never` to work with it.